### ISSUE TITLE
about plot problems on mac or linux

### ISSUE THREAD

**Issue by hackerwei** [mementum/backtrader#49]
Find there are problems when show picutures with plot func on Mac/Linux.
python multitrades.py --plot
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/matplotlib/artist.py", line 61, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/figure.py", line 1139, in draw
    self.canvas.draw_event(renderer)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 1809, in draw_event
    self.callbacks.process(s, event)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/cbook.py", line 562, in process
    proxy(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/cbook.py", line 429, in __call__
    return mtd(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/backtrader/plot/multicursor.py", line 190, in clear
    self.canvas.copy_from_bbox(self.canvas.figure.bbox))
AttributeError: 'FigureCanvasMac' object has no attribute 'copy_from_bbox'
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/matplotlib/backend_bases.py", line 1952, in motion_notify_event
    self.callbacks.process(s, event)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/cbook.py", line 562, in process
    proxy(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/cbook.py", line 429, in __call__
    return mtd(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/backtrader/plot/multicursor.py", line 221, in onmove
    self._update(event)
  File "/usr/local/lib/python2.7/site-packages/backtrader/plot/multicursor.py", line 230, in _update
    ax.draw_artist(line)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/axes/_base.py", line 2320, in draw_artist
    a.draw(self._cachedRenderer)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/artist.py", line 61, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/matplotlib/lines.py", line 737, in draw
    gc = renderer.new_gc()
  File "/usr/local/lib/python2.7/site-packages/matplotlib/backends/backend_macosx.py", line 99, in new_gc
    self.gc.save()
RuntimeError: CGContextRef is NULL

I solve the error by adding code in the plot.py:
 29 from ..utils.py3 import range, with_metaclass
 30 import matplotlib
 31 matplotlib.use('tkagg')
But the pic is not as clear as the original one.

**Comment by mementum** [comment]
Which version of matplotlib are you using?

**Comment by ChrisAllisonMalta** [comment]
I have a similar issue, matplotlib never seems to work on a mac!

I can get general plots to work when I set the backend to Qt4Agg in the matplotlibrc file, similar to what @hackerwei has done above yet when I run cerebro.plot() I get an error:
Unknown property color_cycle

**Comment by mementum** [comment]
- Which version of matplotlib are you using?
  - Self-installed or with a packed distribution? (Anaconda, PythonXY, WinPython)

**Comment by ChrisAllisonMalta** [comment]
I created my environment with Anaconda and then upgraded matplotlib using pip to version 1.5.0

**Comment by mementum** [comment]
The 1.5.0 may be the trouble cause. Still on 1.4.1 and I imagine matplotlib has made some internal changes (the API doesn't seem to be the stablest of all)

I see that WinPython udpated to 1.5.0 just 3 days ago. I guess I'll be able to get to it soon.

**Comment by ChrisAllisonMalta** [comment]
No I don't think it is the 1.5.0 I get the same error actually in 1.4.1 and 1.4.3

Its probably the interface between the plotting library and the mac gui that I need to get working, then I think it will be fine
http://matplotlib.org/faq/virtualenv_faq.html#osx

**Comment by hackerwei** [comment]
sorry for late reply, the version of matplotlib is 1.5.0.

**Comment by mementum** [comment]
I found the time to upgrade to Python distributions with matplotlib 1.5.0 (for Python 2.7, Python 3.4 and Python 3.5)

The charts are shown as before, so it must be down to the virtualenv issue pointed out above by ChrisAllison or the backend.

My development environment is based in WinPython and there the default backend is "TkAgg" (I printed out the value to be sure)

Quoting from matplotlib:

  - The TKAgg backend doesn’t require any external dependencies and is normally always available.

It would/will be then reasonable to force (because these are the default plots produced by the framework) to set the backend to be ``TkAgg`` as done by hackerwei in the original post.

This should avoid conflicts with ``wxagg`` and other backends (which I have not tested)
